src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -146,9 +149,13 @@ public DurableTaskExtension()
             this.HttpApiHandler = new HttpApiHandler(this, logger);
 
 #if !FUNCTIONS_V1
+            this.hostLifetimeService = hostLifetimeService;
+
             // The RPC server is started when the extension is initialized.
             // The RPC server is stopped when the host has finished shutting down.
             hostLifetimeService.OnStopped.Register(this.StopLocalRcpServer);
+#else
+            this.hostLifetimeService = hostLifetimeService ?? new HostLifecycleService();

```

Comment:
I'm pretty sure we can't pass in the default `HostLifecycleService` as a default parameter, as it is not a value type.

I also like using the conditional compilation here, as it doesn't make sense to register `this.StopLocalRpcServer` against a no-op `hostLifetimeService`. The conditional compilation here highlights that V1 and V2 handle this problem differently.

I think the crux of the lack of clarity here is in how I used conditional compilation to change the behavior of `HostLifecycleService`. Instead, Functions V1 should have a conditionally compiled different implementation of `IApplicationLifetimeWrapper`, that is explicitly called `NoOpLifecycleService`. That way, this line changes to 

```csharp
this.hostLifetimeService = new NoOpLifecylceService();
```

which is more clear.

Reply:
Rather than creating a new class, I think it would be sufficiently clear if you just created a static method or property that represented the creation of a no-op instance. For example, using `HostLifecycleService.NoOp` instead of `new HostLifecycleService()`.

Also, I'm not suggesting a default parameter. I'm suggesting an actual parameter value. But I'm not sure I understand why this constructor would even get called in V1. Presumably it should only get called in V2 since only V2 supports DI. I wonder if the right thing to do is just set the default assignment to the no-op lifecycle service - i.e. 

```csharp
private readonly HostLifecycleService hostLifetimeService = HostLifecycleService.NoOp;
```

And then just overwrite it in the ctor for V2 only.

Reply:
So I always get confused by the constructor on line 152, which is V1 only which calls this constructor. I think it is dead code, but it always makes me think that this code path is called in V1, which it isn't. 

I think I addressed most of your feedback, though internally I did have to make a new class for `HostLifeCycleService.NoOp`, as I had no good way of making a noop version of `IApplicationLifecycleService` otherwise.